### PR TITLE
adjust register_host_custom_repo for sca

### PR DESCRIPTION
**Pull request by pondrejk**
If organization is in sca mode, adding  bulk subscriptions to a host will result in 
```
400 Client Error: Bad Request for url: https://<sat>/api/v2/hosts/bulk/add_subscriptions

'{"displayMessage":"The specified organization is in Simple Content Access mode. Attaching subscriptions is disabled","errors":["The       specified organization is in Simple Content Access mode. Attaching subscriptions is disabled"]}
```

This affects a number of tests in rex component, result after change:

```
pytest tests/foreman/cli/test_remoteexecution.py::TestAnsibleREX::test_positive_run_reccuring_job
================================================ test session starts ================================================
platform linux -- Python 3.11.1, pytest-7.2.0, pluggy-1.0.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/pondrejk/Documents/robottelo, configfile: pyproject.toml
plugins: xdist-3.1.0, services-2.2.1, mock-3.10.0, cov-3.0.0, reportportal-5.1.3, ibutsu-2.2.4
collected 1 item                                                                                                    

tests/foreman/cli/test_remoteexecution.py . 
```

### CODE REVIEW

File: tests/foreman/cli/test_remoteexecution.py [hunk 1]
```diff
@@ -551,7 +551,6 @@ def test_positive_run_reccuring_job(self, rex_contenthost):
             }
         )
         result = JobInvocation.info({'id': invocation_command['id']})
-        assert_job_invocation_status(invocation_command['id'], client.hostname, 'queued')
```
Comment (Griffin-Sullivan) [inline tests/foreman/cli/test_remoteexecution.py:554]: Why do we no longer want to assert `queued` here? I see another test in this file that still does this.

**Review by Griffin-Sullivan — APPROVED** [review]
One small comment. If both tests no longer need that assertion for queued, then that helper method will no longer be needed (or maybe we need to use it in other places).

File: robottelo/hosts.py [hunk 2]
```diff
@@ -1770,16 +1767,23 @@ def register_host_custom_repo(self, module_org, rhel_contenthost, repo_urls):
...
-            data={
-                "organization_id": module_org.id,
-                "included": {"ids": [rhel_contenthost.nailgun_host.id]},
-                "subscriptions": [{"id": subscription.id, "quantity": 1}],
-            }
-        )
-        # refresh repository metadata on the host
-        rhel_contenthost.execute('subscription-manager repos --list')
+        # this part is not needed if org is in SCA mode
+        if entities.Organization(id=module_org.id).read().simple_content_access is False:
```
Comment (Gauravtalreja1) [inline robottelo/hosts.py:1773]:
```suggestion
        # Attach subscriptions only if SCA mode is disabled
        if self.api.Organization(id=module_org.id).read().simple_content_access is False:
```

**Review by Gauravtalreja1 — COMMENTED** [review]
Ack, with suggestions

**Review by Gauravtalreja1 — APPROVED** [review]

**Pull request merged**